subsys/mpsl/fem/CMakeLists.txt

```diff
@@ -4,12 +4,13 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-if (CONFIG_MPSL_FEM_NRF21540_GPIO OR CONFIG_MPSL_FEM_SIMPLE_GPIO)
+if (CONFIG_MPSL_FEM_ANY_SUPPORT)
```

Comment:
I'm wondering if this change is beneficial. Consider a user that wants to add their custom FEM implementation outside of `subsys/mpsl/fem`. Most probably such a user will use the existing integration scheme, which implies that `CONFIG_MPSL_FEM_ANY_SUPPORT` will become `y`. In that case, this library will be created and it's only going to contain the common utils.

At the moment, it's non-trivial (or perhaps impossible - not sure) to use this library from an out-of-tree module. Even if we rework this library so that it can be conveniently linked to in CMake, we can't even tell up-front if it's desired to expose it at all - that is going to depend on particular use cases.

Wouldn't it be better to share these utils only among FEM implementations maintained by us? Then this condition should become:
```
if (CONFIG_MPSL_FEM_NRF21540_GPIO OR CONFIG_MPSL_FEM_NRF21540_GPIO_SPI OR CONFIG_MPSL_FEM_SIMPLE_GPIO)
``` 
Opinions are welcome @ankuns @ahasztag.

Reply:
> The OOTM implementation can benefit from this. If it desires it can include mpsl_fem_utils.h.

My point is that the above statement is incorrect. For instance, note that `zephyr_library_include_directories` specifies scope as `PRIVATE`: https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/extensions.cmake#L476-L478, so including `mpsl_fem_utils.h` from outside is not an option. Also, the name of the library is inferred from the directory: https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/extensions.cmake#L390-L393, so there's no obvious way of using CMake's `_link_libraries` syntax - it's unclear what the argument would have to be.

I'm not an expert in CMake, so before writing the above comment I wanted to verify if my understanding is correct. I made an experiment that mimicked the described use case: I removed simple_gpio implementation from `subsys/mpsl/fem` and copied it to a separate directory `subsys/mpsl/new_fem`. This new simple_gpio put in a separate directory with separate Kconfig and CMakeLists.txt acted as a custom FEM implementation. What I found out is that it is _not_ possible to use the common library in its current shape:
```
nrf/subsys/mpsl/new_fem/mpsl_fem_simple_gpio.c:23:10: fatal error: common/include/mpsl_fem_utils.h: No such file or directory
   23 | #include <mpsl_fem_utils.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```
Yes, it _is_ possible to pull in the files directly through CMake using e.g. the following code:
```CMake
if (CONFIG_MPSL_FEM_SIMPLE_GPIO)
    zephyr_library()

    zephyr_library_include_directories(${NRF_DIR}/subsys/mpsl/fem/common/include)
    zephyr_library_sources(${NRF_DIR}/subsys/mpsl/fem/common/mpsl_fem_utils.c)

    zephyr_library_sources_ifdef(CONFIG_MPSL_FEM_SIMPLE_GPIO mpsl_fem_simple_gpio.c)
endif()
```
but AFAIU that's not what you meant.

Assuming all of the above is correct, I see two options now:
* modify the common library so that it can be used easily by other modules;
* expect any custom FEM implementation to provide everything it needs.

In my original comment I suggested going for the second solution, and to make it clear in the code, replacing
```CMake
if (CONFIG_MPSL_FEM_ANY_SUPPORT)
```
with
```CMake
if (CONFIG_MPSL_FEM_NRF21540_GPIO OR CONFIG_MPSL_FEM_NRF21540_GPIO_SPI OR CONFIG_MPSL_FEM_SIMPLE_GPIO)
```

Let me know what you think - perhaps the reasoning above is wrong.

Reply:
Please see if the current solution is OK